### DISCUSSION TITLE
Conditional Log Messages?

### DISCUSSION THREAD

**Discussion by reward-stephen** [artilleryio/artillery#1014] [Artillery]
Anyone had any luck displaying log messages only when something happens?

Is it possible to either:

- Nest `log:` messages within an operation definition, so that the `ifTrue:` setting dictates whether the message is displayed
OR
- Expand the `log:` statement to contain both the message or it's own `ifTrue:` check

At the minute the only alternative seems to be the javascript hooks, which seems a bit overkill for a log message

Thanks!

**Comment by hassy** [comment]
hi @reward-stephen, there's no way to make logging with `log` conditional at the moment, but that's an interesting use case.

Are you mostly interested in logging something if a variable is defined/undefined (e.g. with support for something like `ifTrue`) or something else more advanced too?
